gssapi.go

```diff
@@ -55,7 +55,9 @@ func (m *GSSAPIMechanism) step(challenge []byte) ([]byte, error) {
 	} else if m.negotiationStage == 1 {
 		err := initClientContext(m.context, m.service+"/"+m.host, challenge)
 		if err != nil {
-			log.Fatal(err)
+			if err != gssapi.ErrContinueNeeded {
```

Comment:
I think this would still have the same problem as before. We should a single if with something like:

```
                if err != nil && err != gssapi.ErrContinueNeeded {
			log.Fatal(err)
			return nil, err
		}
``